Modules/Forum/classes/class.ilObjForumGUI.php

```diff
@@ -2706,26 +2694,28 @@ private function getForumObjects()
 
     public function checkUsersViewMode() : void
     {
-        if (!isset($_SESSION['viewmode'])) {
-            $_SESSION['viewmode'] = $this->objProperties->getDefaultView();
+        if (!isset($this->selectedSorting)) {
+            $this->selectedSorting = $this->objProperties->getDefaultView();
         }
 
-        if (isset($_GET['viewmode']) && (int) $_GET['viewmode'] !== (int) $_SESSION['viewmode']) {
-            $_SESSION['viewmode'] = (int) $_GET['viewmode'];
+        if (isset($this->httpRequest->getQueryParams()['viewmode'])
+            && (int) $this->httpRequest->getQueryParams()['viewmode']  !== (int) $this->selectedSorting) {
+            $this->selectedSorting = (int) $this->httpRequest->getQueryParams()['viewmode'];
         }
 
-        if (isset($_GET['action']) && !in_array($_SESSION['viewmode'], [
+        if (isset($_GET['action']) && !in_array($this->selectedSorting, [
```

Comment:
Please use `$this->httpRequest->getQueryParams()` instead of `$_GET`, if possible.

Reply:
We decided, to change the $_GET['viewmode']  to $this->httpRequest->getQueryParams() only. We could change all GET parameters, If needed.

Reply:
We did not, maybe we misunderstood. We **MUST NOT** rely only on HTTP query parameters.

Scenario:

1. Given a user enters the thread view of a forum (list of postings)
2. ... and the user selects a viewmode
3. When the user navigates to (e.g.) a course and then back to the thread view of the forum (list of postings)
4. Then the previous selected viewmode **MUST** be applied.